### ISSUE TITLE
TypeError("'NoneType' object is not iterable")

### ISSUE THREAD

**Issue by s5248** [mosaicml/diffusion#47]
Error:
InstantiationException: Error in call to target 'diffusion.datasets.laion.laion.build_streaming_laion_dataloader':
TypeError("'NoneType' object is not iterable")
full_key: dataset.train_dataset
ERROR:composer.cli.launcher:Rank 0 crashed with exit code 1.
Waiting up to 30 seconds for all training processes to terminate. Press Ctrl-C to exit immediately.
Global rank 0 (PID 34840) exited with code 1
ERROR:composer.cli.launcher:Global rank 0 (PID 34840) exited with code 1

config:
dataset:
  train_batch_size: ${batch_size}
  eval_batch_size: 1 # Should be 8 per device
  train_dataset:
    _target_: diffusion.datasets.laion.laion.build_streaming_laion_dataloader
    remote:
      # Path to object store bucket(s)
    local: /home/diffusion/data/train
      # Path to corresponding local dataset(s)
    batch_size: ${batch_size}
    tokenizer_name_or_path: stabilityai/stable-diffusion-2-base
    caption_drop_prob: 0.1
    resize_size: 512
    drop_last: true
    shuffle: true
    prefetch_factor: 2
    num_workers: 1
    persistent_workers: true
    pin_memory: true
    download_timeout: 300
    num_canonical_nodes: 64

**Comment by Landanjs** [comment]
Apologies for the delay! This might be from an ungraceful termination in the streaming process. Did you have have a previous run that failed? 

Could you try [`streaming.base.util.clean_stale_shared_memory`](https://github.com/mosaicml/streaming/blob/main/streaming/base/util.py#L97)? This should remove any corrupted data
